src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1138 @@
+    private static final String APPS = "rest/system/apps";
+    private static final String SESSIONS = "rest/current/session/sessions";
+
+    private static final String CONTROL_MUTE_VIDEO = "Mute Local Video";
+    private static final String CONTROL_MUTE_MICROPHONES = "Mute Microphones";
+    private static final String CONTROL_AUDIO_VOLUME = "Audio Volume";
+    private static final String CONTROL_REBOOT = "Reboot";
+
+    private static int DEFAULT_CALL_RATE;
+    private static final int MAX_STATUS_POLL_ATTEMPT = 5;
+
+    private final ReentrantLock controlOperationsLock = new ReentrantLock();
+    private long latestControlTimestamp;
+    private ExtendedStatistics localStatistics;
+    private EndpointStatistics localEndpointStatistics;
+
+    private String sessionId;
+
+    /* Grace period for device reboot action. It takes about 3 minutes for the device to get fully
+     * functional after reboot is triggered */
+    private static final int REBOOT_GRACE_PERIOD_MS = 200000;
+
+    /**
+     * Instantiate and object with trustAllCertificates set to 'true' for being able to
+     * communicate with the devices with invalid certificates
+     */
+    public PolycomVideoOS() {
+        setTrustAllCertificates(true);
+    }
+
+    @Override
+    protected void internalInit() throws Exception {
+        super.internalInit();
+
+        Properties properties = new Properties();
+        properties.load(getClass().getResourceAsStream("/adapter.properties"));
+        DEFAULT_CALL_RATE = Integer.parseInt(properties.getProperty("defaultCallRate"));
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * If the call is in progress and another participant is addressed with {@link #CONFERENCES} POST call -
+     * VideoOS Rest API will add the participant as another connection for the existing conference call, without
+     * creating an additional conference call, so it is commonly expected that there's a single conference at most
+     */
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Dialing using dial string: " + dialDevice.getDialString());
+        }
+        ObjectNode request = JsonNodeFactory.instance.objectNode();
+        Integer callSpeed = dialDevice.getCallSpeed();
+
+        request.put("address", dialDevice.getDialString());
+        request.put("rate", (callSpeed != null && callSpeed > 0) ? callSpeed : DEFAULT_CALL_RATE);
+
+        Protocol protocol = dialDevice.getProtocol();
+        if (protocol != null) {
+            request.put("dialType", protocol.name());
+        }
+        ArrayNode response = doPost(CONFERENCES, request, ArrayNode.class);
+        if (response == null) {
+            // Valid response was not received, cannot go further to checking the conference id retrieval
+            throw new IllegalStateException(String.format("Unable to receive response from %s", CONFERENCES));

```

Comment:
Wrong exception. IllegalStateException is about state of the object during which requested operation cannot be performed. E.g. if something asked to be done by object was not initialized yet or was destroyed yet.
At what circumstances response will be null? Would help to determine correct exception type. If there are no suitable exception, just Exception can be used. Sometimes RuntimeException is suitable - if unexpected error happened at run time.